### PR TITLE
add pspuart package

### PR DESCRIPTION
**Pull request by tpimh**

### CODE REVIEW

File: pspuart/PSPBUILD [hunk 1]
```diff
@@ -0,0 +1,28 @@
+pkgname=pspuart
+pkgver=20211106
+pkgrel=1
+pkgdesc="Library for PSP UART hardware communication (ex SioDriver)"
+arch=('mips')
+url="https://github.com/Operation-DITTO/psp-uart-library"
+license=('cc-by-sa-3.0')
```
Comment (sharkwouter) [inline pspuart/PSPBUILD:7]: This does not match the bundled license, it should be cc-by-sa-4.0.

**Review by sharkwouter — CHANGES_REQUESTED** [review]
Thanks for making this PR, always cool to see new libraries. I did find an issue with the license, otherwise this package looks good.

**Comment by tpimh** [comment]
Oops, I tried to use commit hash as pkgver and epoch to increment on every change, but it seems that files with colon in their name can't be uploaded as artifacts.

**Comment by sharkwouter** [comment]
That's odd, that means we cannot use epoch.

**Comment by tpimh** [comment]
There must be a workaround, I'll check. Using it makes sense with packages with no version.

**Comment by sharkwouter** [comment]
You could copy the pkgver function from libpspvram. That takes the commit number and the sha hash to create the version number.

**Comment by tpimh** [comment]
I wonder if it should be called pspuart or libpspuart... Since all the packages are libraries, "lib" prefix seems redundant.

**Comment by sharkwouter** [comment]
Yeah, lib in front of the package name is kinda useless for most packages. Some have it explicitly in their publicly known name.

Thanks for this update, I've merged it.

**Pull request merged**